neptune/mapper.go

```diff
@@ -41,7 +42,22 @@ func (n *NeptuneDB) buildHierarchyNode(v graphson.Vertex, instanceID, dimension
 	}
 	// Fetch new data from the database concerned with the node's children.
 	if res.NoOfChildren > 0 && instanceID != "" {
-		gremStmt := fmt.Sprintf(query.GetChildren, instanceID, dimension, res.ID)
+
+		// Check if order is defined
+		var orderCount int64
+		gremStmt := fmt.Sprintf(query.CountChildrenWithOrder, instanceID, dimension, res.ID)
+		orderCount, err = n.getNumber(gremStmt)
+		if err != nil {
+			return nil, errors.Wrapf(err, "Gremlin query failed: %q", gremStmt)
+		}
+		hasOrder := orderCount > 0
+
+		// query depending on the presence of order property in child nodes
+		if hasOrder {
```

Comment:
You could replace line 56 with:
  if orderCount > 0 {

and remove line 53
(the code will still be obvious to read)

Reply:
yes, this is the only place where we use this bool, so it's really not needed.